x/metoken/keeper/reserves.go

```diff
@@ -0,0 +1,81 @@
+package keeper
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/umee-network/umee/v5/x/metoken"
+)
+
+// RebalanceReserves checks if the portion of reserves is below the desired and transfer the missing amount from
+// x/leverage to x/metoken reserves, or vice versa.
+func (k Keeper) RebalanceReserves() error {
+	rebalancingTime, err := k.getNextRebalancingTime()
+	if err != nil {
+		return err
+	}
+
+	if k.ctx.BlockTime().After(rebalancingTime) {
+		indexes := k.GetAllRegisteredIndexes()
+		for _, index := range indexes {
+			balances, err := k.IndexBalances(index.Denom)
+			if err != nil {
+				return err
+			}
+
+			// if no meToken were minted, there is nothing to check
+			if balances.MetokenSupply.IsPositive() {
+				updatedBalances := make([]metoken.AssetBalance, 0)
+				for _, balance := range balances.AssetBalances {
+					if balance.AvailableSupply().IsPositive() {
+						i, assetSettings := index.AcceptedAsset(balance.Denom)
+						if i < 0 {
+							return sdkerrors.ErrNotFound.Wrapf("asset %s not found in the index", balance.Denom)
```

Comment:
An accepted asset cannot be removed from an index: https://github.com/umee-network/umee/blob/main/x/metoken/README.md?plain=1#L48.

However, we may not want to halt the chain if for some unexpected reason it's not found. Shall I change it to log instead of returning error?

Reply:
log makes sense as long as the module's state isn't left in a weird state (the way a half-executed transaction would mess it up)